src/cmd/deploy.ts

```diff
@@ -1060,6 +1078,99 @@ export default class DefenderDeploy {
     );
   }
 
+  private async deployForkedNetworks(output: DeployOutput<DefenderForkedNetwork>) {
+    const forkedNetworks: ForkedNetworks = this.resources?.['forked-networks'] ?? {};
+    const client = getNetworkClient(this.teamKey!);
+    const retrieveExisting = () => client.listForkedNetworks();
+
+    await this.wrapper<ForkedNetworkRequest, DefenderForkedNetwork>(
+      this.serverless,
+      'Forked Networks',
+      forkedNetworks,
+      retrieveExisting,
+      // on update
+      async (forkedNetwork: ForkedNetworkRequest, match: DefenderForkedNetwork) => {
+        // Warn users when they try to change fields that are not allowed to be updated
+        if (match.name !== forkedNetwork.name) {
+          this.log.warn(
+            `Detected a name change from ${match.name} to ${forkedNetwork.name} for Forked Network: ${match.stackResourceId}. Defender does not currently allow updates to the name once a Forked Network is created. This change will be ignored. To enforce this change, remove this Forked Network and create a new one. Alternatively, you can change the unique identifier (stack resource ID), to force a new creation of the Forked Network. Note that this change might cause errors further in the deployment process for resources that have any dependencies to this network.`,
+          );
+          forkedNetwork.name = match.name;
+        }
+
+        if (match.forkedNetwork !== forkedNetwork['forked-network']) {
+          this.log.warn(
+            `Detected a change from ${match.forkedNetwork} to ${forkedNetwork['forked-network']} for Forked Network: ${match.stackResourceId}. Defender does not currently allow updates to the fork source once a Forked Network is created. This change will be ignored. To enforce this change, remove this Forked Network and create a new one. Alternatively, you can change the unique identifier (stack resource ID), to force a new creation of the Forked Network. Note that this change might cause errors further in the deployment process for resources that have any dependencies to this network.`,
+          );
+          forkedNetwork['forked-network'] = match.forkedNetwork as SupportedNetwork;
```

Comment:
nit: I think `forked-network` is confusing, we could use `forked-from` in a future refactor